From 2058cac12c8b42b7c18ef0ea8755c2243514a79b Mon Sep 17 00:00:00 2001
From: ArenM <aren@peacevolution.org>
Date: Wed, 27 Apr 2022 23:57:38 -0400
Subject: [PATCH sxmo-utils] Systemd compatibility

Sxmo's default configuration attempts to start services that systemd
expects to manage.
---
 configs/default_hooks/sxmo_hook_start.sh | 19 -------------------
 scripts/core/sxmo_mmsdconfig.sh          |  3 +++
 scripts/core/sxmo_vvmdconfig.sh          |  3 +++
 3 files changed, 6 insertions(+), 19 deletions(-)

diff --git a/configs/default_hooks/sxmo_hook_start.sh b/configs/default_hooks/sxmo_hook_start.sh
index d7256f3..ab1a6db 100755
--- a/configs/default_hooks/sxmo_hook_start.sh
+++ b/configs/default_hooks/sxmo_hook_start.sh
@@ -16,16 +16,6 @@ while ! superctl status > /dev/null 2>&1; do
 	sleep 0.5
 done
 
-# Load our sound daemons
-
-if [ "$(command -v pulseaudio)" ]; then
-	superctl start pulseaudio
-elif [ "$(command -v pipewire)" ]; then
-	# pipewire-pulse will start pipewire
-	superctl start pipewire-pulse
-	superctl start wireplumber
-fi
-
 # Periodically update some status bar components
 sxmo_hook_statusbar.sh all
 sxmo_daemons.sh start statusbar_periodics sxmo_run_aligned.sh 60 \
@@ -84,15 +74,6 @@ superctl start sxmo_soundmonitor
 
 # Set a background image: edit ~/.config/sxmo/sway or ~/.config/sxmo/xinit
 
-# mmsd and vvmd
-if [ -f "${SXMO_MMS_BASE_DIR:-"$HOME"/.mms/modemmanager}/mms" ]; then
-	superctl start mmsd-tng
-fi
-
-if [ -f "${SXMO_VVM_BASE_DIR:-"$HOME"/.vvm/modemmanager}/vvm" ]; then
-	superctl start vvmd
-fi
-
 # add some warnings if things are not setup correctly
 deviceprofile="$(command -v "sxmo_deviceprofile_$SXMO_DEVICE_NAME.sh")"
 
diff --git a/scripts/core/sxmo_mmsdconfig.sh b/scripts/core/sxmo_mmsdconfig.sh
index 035ab93..36531e1 100755
--- a/scripts/core/sxmo_mmsdconfig.sh
+++ b/scripts/core/sxmo_mmsdconfig.sh
@@ -7,6 +7,9 @@
 # shellcheck source=scripts/core/sxmo_common.sh
 . "$(dirname "$0")/sxmo_common.sh"
 
+# on arch we manage mmsd with systemd instead of superd
+alias superctl='systemctl --user'
+
 set -e
 
 MMS_BASE_DIR="${SXMO_MMS_BASE_DIR:-"$HOME"/.mms/modemmanager}"
diff --git a/scripts/core/sxmo_vvmdconfig.sh b/scripts/core/sxmo_vvmdconfig.sh
index ab14e4e..ae76ed0 100755
--- a/scripts/core/sxmo_vvmdconfig.sh
+++ b/scripts/core/sxmo_vvmdconfig.sh
@@ -9,6 +9,9 @@
 
 set -e
 
+# on arch we manage mmsd with systemd instead of superd
+alias superctl='systemctl --user'
+
 VVM_BASE_DIR="${SXMO_VVM_BASE_DIR:-"$HOME"/.vvm/modemmanager}"
 VVMCONFIG="$VVM_BASE_DIR/vvm"
 
-- 
2.39.0

